### ISSUE TITLE
Target and DependsOn does not respect order

### ISSUE THREAD

**Issue by Werebunny** [nuke-build/nuke#90]
Nuke worked great for my project, but I encountered a few issues.

## Expected Behavior
Specifying Test+Coverage or DependsOn(Test, Coverage) should execute Test build first, then Coverage build.

## Current Behavior
Coverage runs first, then Test. It appears to be following alphabetical order.

## Steps to Reproduce
1. Create project or clone [CreateAndFake](https://github.com/CreateAndFake/CreateAndFake).
2. Run a combo build contrary to alphabetical order, such as './build/build Tests+Coverage' for CreateAndFake.

## Additional Information
This greatly reduces my ability to tie together builds. Like on AppVeyor, I'd like to create a target that depends on Test and Coverage, but it's runs them out of order. And because Test depends upon Clean, where Coverage doesn't, Test deletes my coverage results.

Instead, my server script first calls build Test, then calls build Coverage, since build Test+Coverage has this issue as well.

**Comment by matkoch** [comment]
Thanks for the posting this issue.

A `DependsOn` call does not express any dependencies between the targets passed as arguments. If you want `Test` to be executed before `Coverage`, then you need to call `DependsOn(Test)`. I've looked at your build and think this should work. Probably needs some conditions using `OnlyWhen`.

I'm closing this as *won't fix* since this is the expected behavior.

**Comment by Werebunny** [comment]
Making Coverage depend upon Test in my situation, makes any time I want to check coverage myself add another minute of irrelevant building and testing.

Let me give another scenario to clarify here:

Say I have a Target Clean, as you would expect.

And then I have a Target Build, this is instead of the normal name you gave it, Compile.

Now, say I don't want Build to depend upon Clean. I want the option to clean the output directory, but I want Clean and Build to remain separate options to use as the user wants.

And then I come across your supposedly neat feature of chaining targets. So, running "build.cmd Clean+Build" makes it so easy to run them together if I want!

Or, say I make another target, such as one to run on a build server, than I want to depend upon Clean, then Build, then maybe something else, and so forth.

Both of these fail. Specifying build Clean+Build forces Build to run first, because you for some reason changed the order to follow alphabetical order instead of the way I specified it.

And I can't make a Target that depends upon Clean and Build for the same reason.

This is a pretty bad bug, not a feature. While you claim you support multiple Target calls, it's broken in this way, so you only support chaining multiple build paths if it's in alphabetical order. In more generic terms:

A -> B

Y -> X

You cannot build X+B. You can only build B+X. And this is bad when the B path does something that can mess up X.

**Comment by matkoch** [comment]
Like I said, this is expected behavior. The order of targets specified in a `DependsOn` call does not guarantee any execution order. As far as I know, we also don't sort targets or anything. The fact they get executed in alphabetical order is purely random behavior. Eventually, we will implement support for parallel target execution, and in this case, your `Test` and `Coverage` will be executed in parallel in fact.

You said it yourself, that B can do something that messes X up. This is a clear indicator that there is a dependency between those two targets. I think you should consider using `-skip` instead. There is also a related PR #92 to be merged soon.

**Comment by arodus** [comment]
As matkoch already said it's the expected behavior but we deciced to add the possibility to order targets without hard dependencies(#93) instead.

**Comment by Werebunny** [comment]
That'll work great, thanks.
